### ISSUE TITLE
Poor test performance of TRPO on Hopper-v2

### ISSUE THREAD

**Issue by HareshMiriyala** [hill-a/stable-baselines#581]
I trained a TRPO agent with standard hyperparameters on the Hopper-v2 environment in mujoco for 10M timesteps. At the end of the training iterations, i noticed that the EpRewMean is around 2.39e+03. However, when I deploy this policy on the environment and average the returns over 100 trajectories, I get an average return of 1300 which is not so great in the Hopper domain. I have seen papers that achieve 3500 return on this environment. Can someone help me understand what's going on ?
 
 * Nvidia RTX GPU
 * Python version : 3.6
 * Tensorflow version : 1.14.0

**Comment by araffin** [comment]
Hello,
Please fill the issue template completely. Did you use the rl zoo for training? Did you use VecNormalize?

**Comment by HareshMiriyala** [comment]
Thanks,  I have tried using both the default parameters and also the hyperparameters from rl zoo. Neither improved test performance. Also, I don't use VecNormalize. Just DummyVecEnv

**Comment by HareshMiriyala** [comment]
I will try using VecNormalize to see if that improves test performance.

**Comment by araffin** [comment]
It seems that your results match the benchmark: http://htmlpreview.github.io/?https://github.com/openai/baselines/blob/master/benchmarks_mujoco1M.htm
And you may have problem loading the trained policy, I would really recommend using the zoo.

**Comment by araffin** [comment]
Update:
I successfully trained (one seed) an agent using TRPO with those hyperparameters:

```yaml
Hopper-v2:
  env_wrapper: utils.wrappers.TimeFeatureWrapper
  n_timesteps: !!float 2e6
  policy: 'MlpPolicy'
  timesteps_per_batch: 2048
  max_kl: 0.005
  cg_iters: 15
  cg_damping: 0.1
  entcoeff: 0.01
  gamma: 0.99
  lam: 0.95
  vf_iters: 5
  vf_stepsize: !!float 1e-3
```

At the end of training, the stochastic policy is around 2000 but you get a reward around 3600 if you use a deterministic policy for testing.

Closing this issue then.

**Comment by HareshMiriyala** [comment]
Thanks @araffin, I just realized i didn't set deterministic=True when predicting actions using my trained agent. I get around 3600, which is what I needed :)

**Comment by xkianteb** [comment]
@araffin I am not getting 3600 from the pre-trained trpo agent in rl-baselines-zoo.

**Comment by araffin** [comment]
the one from the zoo is a different environment, using pybullet instead of mujoco
